### PR TITLE
CARBG 44: Research match id and player id handling on lobbys

### PR DESCRIPTION
**Pull request by AdolfoTaD**
- Modified the React lobby to function with our customized client and integrated it with CartesifyTransport.
- Ensured all functionalities such as creating, joining, leaving a game, etc., are working correctly.
- The React lobby previously used the default client from the boardgame repository, but now it has been updated to use our modified client and transport.

### CODE REVIEW

File: examples/react-web/server.js [hunk 1]
```diff
@@ -6,15 +6,39 @@
...
 const server = Server({
-  games: [TicTacToe, Chess],
-  origins: [Origins.LOCALHOST],
+  games: [TicTacToe],
+  origins: ['http://localhost:1234', 'http://localhost:3000'],
+});
+
+server.app.use(
+  cors({
+    // origin: ['http://localhost:1234', 'http://localhost:3000'],
```
Comment (fgregoribattista) [inline examples/react-web/server.js]: can remove this comment?

File: examples/react-web/server.js [hunk 2]
```diff
@@ -6,15 +6,39 @@
...
+    origin: '*',
+    // methods: ['GET', 'POST'],
```
Comment (fgregoribattista) [inline examples/react-web/server.js]: and this

File: examples/react-web/src/lobby/lobby.js [hunk 3]
```diff
@@ -1,37 +1,28 @@
...
-GameChess.minPlayers = GameChess.maxPlayers = 2;
 
-const hostname = window.location.hostname;
-const importedGames = [
-  { game: GameTicTacToe, board: BoardTicTacToe },
-  { game: GameChess, board: BoardChess },
-];
+const serverURL = 'http://localhost:8000';
+const nodeURL = 'http://localhost:8080';
+const dappAddress = '0xab7528bb862fB57E8A2BCd567a2e929a0Be56a5e';
```
Comment (fgregoribattista) [inline examples/react-web/src/lobby/lobby.js:12]: its ok this address in const?
Change in the future? cold be a env var?

File: examples/react-web/src/tic-tac-toe/authenticated.js [hunk 4]
```diff
@@ -8,146 +8,68 @@
...
 import request from 'superagent';
 
-const hostname = window.location.hostname;
 const App = Client({
   game: TicTacToe,
   board: Board,
   debug: false,
-  multiplayer: SocketIO({ server: `${hostname}:8000` }),
+  multiplayer: CartesiMultiplayer({
+    server: 'http://localhost:8000',
```
Comment (fgregoribattista) [inline examples/react-web/src/tic-tac-toe/authenticated.js:22]: its ok this values fixed?

File: examples/react-web/src/tic-tac-toe/multiplayer.js [hunk 5]
```diff
@@ -1,37 +1,44 @@
...
+    return <div>Loading...</div>;
+  }
+
+  const App = Client({
+    game: TicTacToe,
+    board: Board,
+    debug: false,
+    multiplayer: CartesiMultiplayer({
+      server: 'http://localhost:8000',
+      dappAddress: '0xab7528bb862fB57E8A2BCd567a2e929a0Be56a5e',
```
Comment (fgregoribattista) [inline examples/react-web/src/tic-tac-toe/multiplayer.js:21]: same here

File: examples/react-web/src/tic-tac-toe/spectator.js [hunk 6]
```diff
@@ -1,43 +1,48 @@
...
-        Spectator
+  if (!signer) {
+    return <div>Loading...</div>;
+  }
+
+  const App = Client({
+    game: TicTacToe,
+    board: Board,
+    debug: false,
+    multiplayer: CartesiMultiplayer({
```
Comment (fgregoribattista) [inline examples/react-web/src/tic-tac-toe/spectator.js:19]: and here!

File: examples/typescript/src/App.ts [hunk 7]
```diff
@@ -18,108 +26,136 @@ interface State {
...
-        const id = 3 * i + j;
-        cells.push(`<td class="cell" data-id="${id}"></td>`);
-      }
-      rows.push(`<tr>${cells.join('')}</tr>`);
-    }
+  try {
+    await window.ethereum.request({ method: 'eth_requestAccounts' });
+  } catch (error) {
+    alert(
+      'Failed to connect to MetaMask. Please ensure MetaMask is unlocked and try again.'
```
Comment (fgregoribattista) [inline examples/typescript/src/App.ts:56]: please put this msg in const var

File: examples/typescript/src/App.ts [hunk 8]
```diff
@@ -18,108 +26,136 @@ interface State {
...
+  const provider = new BrowserProvider(window.ethereum);
+  const signer = await provider.getSigner();
+  let matchID = '';
+
+  const createMatchButton = document.createElement('button');
+  createMatchButton.textContent = 'Create Game';
+  createMatchButton.addEventListener('click', async () => {
+    matchID = await createNewMatch(lobbyClient, gameName, 2);
+    if (!matchID) {
+      alert('Failed to create match. Please try again.');
```
Comment (fgregoribattista) [inline examples/typescript/src/App.ts:89]: same here

File: examples/typescript/src/App.ts [hunk 9]
```diff
@@ -18,108 +26,136 @@ interface State {
...
       return;
     }
-    const cells = this.rootElement.querySelectorAll('.cell');
-    cells.forEach((cell) => {
-      const cellId = parseInt((cell as HTMLElement).dataset.id!);
-      const cellValue = state.G.cells[cellId];
-      (cell as HTMLElement).textContent = cellValue !== null ? cellValue : '';
+
+    const matchCreatedMsg = document.createElement('p');
+    matchCreatedMsg.textContent = `Game created with Match ID: ${matchID};`;
```
Comment (fgregoribattista) [inline examples/typescript/src/App.ts:94]: please put this all 'magic' words in a separate file, like a json/const file, so if in the future need to change some of this text, only have to change that file

File: examples/typescript/src/Lobby.ts [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+  ctx: {
+    gameover?: {
+      winner?: string;
+    };
+    currentPlayer: string;
+  };
+  matchID: string;
+}
+
+const DAPP_ADDRESS = '0xab7528bb862fB57E8A2BCd567a2e929a0Be56a5e';
```
Comment (fgregoribattista) [inline examples/typescript/src/Lobby.ts:23]: same here, please put these values put in const file

Comment (fgregoribattista) [reply examples/typescript/src/Lobby.ts:23]: or env var.

File: examples/typescript/src/Lobby.ts [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+    const lobbyClient = new LobbyClient({
+      server: SERVER,
+      nodeUrl: NODE_URL,
+      dappAddress: DAPP_ADDRESS,
+      signer: signer,
+    });
+
+    return lobbyClient;
+  } catch (error) {
+    console.error('Error initializing LobbyClient in Lobby.ts:', error);
```
Comment (fgregoribattista) [inline examples/typescript/src/Lobby.ts:50]: this its ok?
an error occurs, I send a message and the error doesn't scale? it is lost?

File: examples/typescript/src/Lobby.ts [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+  }
+}
+
+export async function listAvailableGames(
+  lobbyClient: LobbyClient
+): Promise<void> {
+  try {
+    const games: string[] = await lobbyClient.listGames();
+  } catch (error) {
+    console.error('Error listing games in Lobby.ts:', error);
```
Comment (fgregoribattista) [inline examples/typescript/src/Lobby.ts:61]: same here

File: examples/typescript/src/Lobby.ts [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+      listItem.textContent = `Match ID: ${
+        match.matchID
+      }, Created At: ${new Date(
+        match.createdAt
+      ).toLocaleString()}, Status: ${isGameOver}, Winner: ${winner}`;
+
+      matchesListElement.appendChild(listItem);
+    });
+  } catch (error) {
+    console.error(`Error listing matches for ${gameName}:`, error);
```
Comment (fgregoribattista) [inline examples/typescript/src/Lobby.ts:91]: and here

**Review by fgregoribattista — COMMENTED** [review]
don't finish checking the PR, it's huge.
Please check the error handling, there are many files where in a catch a log is issued and then the error is lost.
Also check the values of ser, address and server/app routes are set and should change according to the environment.

File: src/server/api.ts
Comment (jm-aragonpaz) [inline src/server/api.ts:1]: There are a lot of console.logs... should be deleted or replaced

**Review by jm-aragonpaz — APPROVED** [review]
LGTM

**Pull request merged**